runtime/src/extensions.rs

```diff
@@ -99,62 +95,86 @@ impl TryFrom<u16> for v0::FuncId {
 	}
 }
 
+mod utils {
+	use super::*;
+
+	pub fn dispatch_call<T, E>(env: &mut Environment<E, BufInBufOutState>, call: <T as SysConfig>::RuntimeCall) -> Result<(), DispatchError>
```

Comment:
Why pub? Everything is Rust is private by default, so overexposing functions when never required by an external crate is unneccessary imo. Consider pub(crate) or pub(super) to only grant 'just enough' access rather than overexposing. The utils module is indeed private, but if someone were to mistakenly overexpose that then suddenly all internal functions are publicly accessible.

https://doc.rust-lang.org/reference/visibility-and-privacy.html

Reply:
The module utils is private, but the function needs to be public within the utils module

Reply:
pub(super) was the right one!